CondTools/RunInfo/interface/BTransitionAnalyzer.h

```diff
@@ -0,0 +1,72 @@
+#ifndef BTRANSITIONANALYZER_H
+#define BTRANSITIONANALYZER_H
+
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CondCore/CondDB/interface/Utils.h"
+#include "CondCore/DBOutputService/interface/PoolDBOutputService.h"
+#include "CondFormats/RunInfo/interface/RunInfo.h"
+#include "CondFormats/DataRecord/interface/RunSummaryRcd.h"
+
+namespace cond {
+  template<class T, class R>
+  class BTransitionAnalyzer: public edm::one::EDAnalyzer<edm::one::WatchRuns, edm::one::SharedResources> {
+  public:
+    BTransitionAnalyzer( const edm::ParameterSet& pset ):
+      m_currentThreshold( pset.getUntrackedParameter<double>( "currentThreshold", 18000. ) ) {
+      usesResource( "PoolDBOutputService" );
```

Comment:
@ghellwig sorry for late reply.
As discussed few minutes ago at the AlCaDB meeting, the code here was originally developed in 7.4.X, where `PoolDBOutputService` was *not* thread safe.
According to the [API description](https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkOneModuleInterface)

> The system makes sure that only one module which has declared a usage of a specific resource will ever be run at a time. That is, the system will serialize all calls to all modules which declare that resource.

In the online context of the O2O, a `cmsRun` job will call only this module (apart from input sources), so, even if formally the performance penalty is introduced, operationally it should not be visible. Besides this, in that context you care more of the correctness of the database transactions than of the performance of job: if you have a fast job potentially writing wrong IOV, this is a big problem.

Anyhow, if also @Dr15Jones agrees that the `usesResource` is not needed, I can put a PR removing it.

HTH

P.S.: you were also saying another observation at the meeting, but I did not catch it. Feel free to post it  here.